src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final int side;
+    private String type;
+
+    public Square(String color, int side) {
+        super(color);
+        this.side = side;
+        this.setType("square");
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    @Override
+    public double getArea() {
+        return side * side;
+    }
+
+    @Override
+    public void printInfo() {
+        System.out.println("Figure: " + getType() + ", area: " + getArea() + " units, side: " + side
```

Comment:
You can have a direct access to this field as it is in the save class. But i think it could be moved to the parent class Figure because this field is common for each class of this type.
```suggestion
        System.out.println("Figure: " + type + ", area: " + getArea() + " units, side: " + side
```

Reply:
Okay, let's leave it as it is